MarkBind/vue-strap
# Upgrade Vue to v2.0.0

**What is the purpose of this pull request? (put "X" next to an item, remove the rest)**

• [X] Other, please explain: Migration works

Follows MarkBind/markbind#245

**What is the rationale for this request?**
Migration to Vue 2.0 as part of the major migration to bootstrap-vue.

**What changes did you make? (Give an overview)**
Ensure that all components are actually compileable on Vue 2.0.

**Provide some example code that this change will affect:**
NIL

**Is there anything you'd like reviewers to focus on?**
NIL

**Testing instructions:**
1. `npm install` to update Vue to v2 (when you are off this branch, remember to downgrade back to v1).
2. `npm run build`
3. Copy `dist/vue-strap.min.js` to your MarkBind repository (`assets/js`).
4. Copy `node_modules/vue/dist/vue.min.js` to your MarkBind repository (`assets/js`) to upgrade to Vue 2.0.
5. Create a new MarkBind website (`markbind init`).
6. Try a component that works without additional migration efforts (i.e. `TipBox`, `pic`)*. For example:
```html
<tip-box type="wrong">
    wrong
</tip-box>

<pic src="https://vuejs.org/images/logo.png" alt="Logo">
    Logo For Vue.js
</pic>
```
7. `markbind serve` and check that it still works for these components (otherwise ensure that you did step 4).

 *These two will not cause any additional errors after upgrading to vue 2.0, and so they work. The rest of the components are not migrated yet, they only have their compilation errors fixed but they still use some outdated attributes and features, so they will have runtime errors.
